src/lib/helpers.ts

```diff
@@ -15,6 +15,51 @@ export function uuid() {
 
 export const isBrowser = () => typeof document !== 'undefined'
 
+const localStorageWriteTests = {
+  tested: false,
+  writable: false,
+}
+
+/**
+ * Checks whether localStorage is supported on this browser.
+ */
+export const supportsLocalStorage = () => {
+  if (!isBrowser()) {
+    return false
+  }
+
+  try {
+    if (typeof globalThis.localStorage !== 'object') {
+      return false
+    }
+  } catch (e) {
+    // DOM exception when accessing `localStorage`
+    return false
+  }
+
+  if (localStorageWriteTests.tested) {
```

Comment:
When will this be true?

Reply:
It's set right below when the write test finishes.